lib/Service/PlatformService.php

```diff
@@ -117,13 +116,17 @@ public function getPlatforms(): array {
 		foreach ($this->platforms as $wrapper) {
 			$class = $wrapper->getClass();
 			try {
-				$platform = OC::$server->query((string)$class);
-				if ($platform instanceof IFullTextSearchPlatform) {
-					$wrapper->setPlatform($platform);
-					$platforms[] = $wrapper;
+				$platform = OC::$server->get((string)$class);
+				if (!($platform instanceof IFullTextSearchPlatform)) {
+					$this->miscService->log($class . ' does not implement ' . IFullTextSearchPlatform::class);
+					continue;
 				}
-			} catch (QueryException $e) {
+				
+				$wrapper->setPlatform($platform);
+				$platforms[] = $wrapper;
+			} catch (ContainerExceptionInterface $e) {
 				/** we cycle */
+				$this->miscService->log($e->getMessage());
```

Comment:
Maybe also logging the exception trace would be helpful here, but cannot tell how that works with the custom log method.